pkg/kubelet/cri/remote/remote_runtime.go

```diff
@@ -1149,3 +1149,50 @@ func (r *remoteRuntimeService) ReopenContainerLog(containerID string) (err error
 	klog.V(10).InfoS("[RemoteRuntimeService] ReopenContainerLog Response", "containerID", containerID)
 	return nil
 }
+
+// CheckpointContainer triggers a checkpoint of the given CheckpointContainerRequest
+func (r *remoteRuntimeService) CheckpointContainer(options *runtimeapi.CheckpointContainerRequest) error {
+	klog.V(10).InfoS(
+		"[RemoteRuntimeService] CheckpointContainer",
+		"options",
+		options,
+	)
+	if options == nil {
+		return errors.New("CheckpointContainer requires non-nil CheckpointRestoreOptions parameter")
+	}
+	if !r.useV1API() {
+		return errors.New("CheckpointContainer is only supported in the CRI v1 runtime API")
+	}
+
+	ctx, cancel := func() (context.Context, context.CancelFunc) {
+		if options.Timeout != 0 {
+			// If the user requested a timeout we use that timeout.
+			return getContextWithTimeout(time.Duration(options.Timeout) * time.Second)
+		}
+		// No timeout requested means we fall back to no time out.
+		return getContextWithCancel()
```

Comment:
I think we should default to some value when none is provided.

Reply:
I am happy to implement whatever is seen as useful. The current no timeout implementation was based on @mikebrow suggestion.

Reply:
Or you could throw a zero timeout value is not supported error?

Reply:
https://github.com/kubernetes/kubernetes/pull/104907#discussion_r900821827

Reply:
Yeah, we can error on negative value here :+1:

Reply:
So, after chatting with @mrunalp and @mikebrow I changed it so that `0` means use CRI default and anything `>0` means use that user-specific timeout. Any negative values are an error.